llama.cpp

```diff
@@ -18714,6 +18952,28 @@ static int32_t llama_chat_apply_template_internal(
         if (add_ass) {
             ss << "<|start_header_id|>assistant<|end_header_id|>\n\n";
         }
+    } else if (tmpl == "chatglm3" ||
+        (tmpl.find("add_generation_prompt") != std::string::npos &&
+        tmpl.find("for message in messages") != std::string::npos &&
+        tmpl.find("loop.first") != std::string::npos)) {
```

Comment:
These conditions are redundant and mostly clash with other chat templates. I'd suggest:

```
tmpl == "chatglm3" || tmpl.find("[gMASK]sop") != std::string::npos
```

Source: https://huggingface.co/THUDM/chatglm3-6b-128k/blob/main/tokenizer_config.json